CodingStandard.md

```diff
@@ -0,0 +1,1425 @@
+code base should be justified or resolved as soon as possible.
+
+**File Naming**
+
+C++ code should adhere to these naming conventions:
+
+  - All source code filenames should be lower case.
+
+  - C++ files should use the .cpp and .h extensions (.hpp extension
+    should not be used).
+
+  - Header files that are #included in the body of another file (e.g. to define
+    a reusable lookup table) should use the -inline.h suffix.
+
+  - All C++ files associated with a design package should be prefixed with the
+    package namespace name followed by an underscore. For example if `nav` is a
+    namespace representing the Navigation package, then
+    `nav_localize.h` and `nav_localize.cpp` are files belonging to the Navigation 
+
+package.
+
+  - Files should be named with the primary entity they contain. Example:
+    `nav_localize.h` contains declaration of class `nav::Localize`.
+    Exceptions are allowed when a file contains an assortment of different
+    types or entities.
+
+  - Files defining an abstract base class should end in `_interface.h`, rather
+    than begin with `i`.
+
+
+**Identifier Naming**
+
+C++ code should adhere to the naming conventions below to promote consistency
+throughout the project.
+```
+  - Pre-Processor Macros: `UPPER_CASE`
+  - Class, Struct, Union, and Enum Types: `CamelCase`
+  - Functions and Class Methods: `lower_case`
+  - Compile-time constants (const, constexpr, enumerators, etc.): `kCamelCase`
+  - Variables and Function Parameters: `lower_case`
+  - constexpr Methods and Functions: `lower_case`
+  - Non-const struct member variables: `lower_case`
+  - Non-const class member variables: `lower_case_` (incl. trailing undercore)
+  - Const member variables: `kCamelCase`
+  - Template type parameters: `CamelCase`
+  - Template value parameters: `kCamelCase`
+  - Namespaces (4 characters max) : `lower_case`
+
+
+- Get methods should be named for the variable they return, and set methods
+  should be the variable name prefixed with `set_`.
+
+- Abstract base classes (interfaces) should be prefixed with I, as in
+  `IContainer`.
+
+- Acronyms and abbreviations should be avoided, unless they are well-known and
+  commonly encountered throughout the project.
+
+- Acronyms in identifiers should be treated as words, for example,
+  `SidStarApproachRecord` as a class name and `sid_star_approach_record` as a
+  variable name.
+
+- Hungarian naming convention or any encoding of names to imply type information
+  should not be used.  This includes prefixing `p` for pointers, `m` for member
+  variables, `s` for static variables, etc.
+```
+[These are the standards we were using at UASC, and I'm not particularly attached
+to them. Specifically, I wouldn't mind having a convention that differentiates
+between public and private members and methods, and I definitely dislike the 
+trailing `_` they use for class members.  What do you guys like?]
+
```

Comment:
things I dislike
m_string
my_string
myString

I prefer
longer_names_with_underscores
or 
camelTypenames

but I'll do whatever we go with.

Reply:
I'm actually a big fan of m_xxx for member variables. Especially those with names that would be nice to have for parameters.

Reply:
I double support for m_xxx for private member variables. I think it makes the code more self documenting (no need to go into the header to determine the source of a m_xxx variable).

Reply:
Yes, private is the keyword here.